syntax/go.vim

```diff
@@ -135,8 +135,8 @@ syn region      goBlock             start="{" end="}" transparent fold
 syn region      goParen             start='(' end=')' transparent
 
 " Integers
-syn match       goDecimalInt        "\<\d\+\([Ee]\d\+\)\?\>"
-syn match       goHexadecimalInt    "\<0x\x\+\>"
+syn match       goDecimalInt        "\<[0-9_]\+\([Ee]\d\+\)\?\>"
```

Comment:
You don't want this to match leading underscores like `_42` though, right? That's an identifier per https://golang.org/ref/spec#Integer_literals. Probably want to change that part of the pattern to `[1-9][0-9_]*`.